modules/ROOT/pages/_partials/proc_self-signed-certs.adoc

```diff
@@ -1,24 +1,23 @@
 //[[using-self-signed-certificates-in-mobile-clients]]
 = Using Self-Signed Certificates in Mobile Apps
 
-Throughout the development lifecycle of a mobile apps, a common
+Throughout the development lifecycle of a mobile app, a common
 requirement is to integrate and connect with back-end
-services in a secure manner. However in order for the client
-device to connect over SSL/TLS, it needs to trust the root
-certificate authority. This cert must be installed on the client device
-to ensure that the client trusts server certificates that are signed by
-the certificate authority.
+services in a secure manner. This is achieved using SSL/TLS. 
 
-When the certificate uses a trusted root certificate authority the cert is installed automatically.
-In a development environment, such as a local OpenShift cluster which uses
-self-signed certificates, you must extract the root certificate from the cluster, install it on the device and mark it as trusted.
+However, in order for the client device to connect over SSL/TLS, it needs to trust the
+certificates used by the back-end services, which are signed by a certificate authority. Today most of the client devices will have a list of well-known and trusted certificate authorities pre-installed and it will allow the client devices to verify the certificates used by the back-end services.
+
+However, this normally doesn't work in a development environment, such as a local OpenShift cluster as it uses self-signed certificates which are not signed by any of the trusted certificate authorties. This means by default the client devices won't be able to establish secure connections with the back-end services that are running on a local OpenShift cluster.
+
+To work around the problem, you must manually extract the root certificate from the cluster, install it on the device and mark it as trusted.
 
 [[extracting-root-ca]]
```

Comment:
I am wondering if this is the easiest way to get the cert trusted. How many devices would support the approach of just opening the URL of the local OpenShift cluster from the device browser and following the instructions presented by the browser to add the certificate. Pretty sure this works on Android, but not sure about the other platforms.

Thoughts?

Reply:
I don't think there are a lot of devices support this approach. I tried it on my own android phone and even if I trusted the web console in the browser, it still doesn't work for me in the app.  Also from Chrome, I don't see there is an option to allow to download the CA file and save it on the device. Does it work on your phone?

Reply:
OK. Let's leave it as is for now so.